pkg/stanza/operator/helper/encoding.go

```diff
@@ -14,48 +14,44 @@ import (
 	"golang.org/x/text/transform"
 )
 
-// NewBasicConfig creates a new Encoding config
+// NewEncodingConfig creates a new Encoding config
 func NewEncodingConfig() EncodingConfig {
 	return EncodingConfig{
 		Encoding: "utf-8",
 	}
 }
 
-// EncodingConfig is the configuration of a Encoding helper
+// EncodingConfig is the configuration of an Encoding helper
 type EncodingConfig struct {
 	Encoding string `mapstructure:"encoding,omitempty"`
 }
 
-// Build will build an Encoding operator.
-func (c EncodingConfig) Build() (Encoding, error) {
```

Comment:
This too

Reply:
Do we have documentation on steps to deprecate? If it's too large of an effort, then it might be better to leave it as is and add a warning to the function godoc that it could result in concurrency issues. I could also just add new functions instead of deprecating old ones.

Reply:
We just need a comment like this: https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/0454100f244052c42b968ad4c15313474a49e184/receiver/simpleprometheusreceiver/receiver.go#L55

Otherwise, we're just allowing the struct to exist as part of the package for another version or two. We don't have to use it ourselves, but this way if others are using it they can still compile while they work out how to migrate.